ws4redis/static/js/ws4redis.js

```diff
@@ -107,4 +135,33 @@ function WS4Redis(options, $) {
 		return ws.readyState;	
 	};
 	
+	function is_connecting() {
+		return ws.readyState == 0;	
+	}
+	
+	function is_connected() {
+		return ws.readyState == 1;	
+	}
+	
+	function is_closing() {
+		return ws.readyState == 2;	
+	}
+
+	function is_closed() {
+		return ws.readyState == 3;	
+	}
+	
+	
+	this.close = function () {
+		clearInterval(heartbeat_interval);
+		must_reconnect = false;
+		if (!is_closing() || !is_closed()) {
+			ws.close();
+		}
+	}
+	
+	this.is_connecting = function () { is_connecting(); } 
```

Comment:
this means that you actually want to hide the boolean return value, if the state is correct? Otherwise it would be simpler to just write:

```
 this.is_connecting = is_connecting;
```